common/logger/Cargo.toml

```diff
@@ -7,5 +7,6 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-env_logger = "0.6"
 log = "0.4"
+log4rs = "0.8"
+serde_json = "1.0"
```

Comment:
newline

Reply:
> newline

I wondered why newline is necessary, did some search and found this <https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline>.
 Pretty interesting.